ServantHDBC/src/Api/Tenant.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators #-}
+
+
+module Api.Tenant (TenantAPI, tenantHandlers) where
+
+import Control.Monad.Trans.Except
+import Servant
+
+type Tenant = String
+type TenantAPI = "tenants" :> 
+                         ("new" :> Post '[JSON] Tenant
+                          :<|> Capture "x" Int :> Get '[JSON] Tenant
+                          :<|> Capture "x" Int :> "activate" :> Post '[JSON] Tenant)
+
+tenantNew :: ExceptT ServantErr IO Tenant
```

Comment:
how does one capture the incoming JSON of a POST in a type-safe manner? Shouldn't this type signature be something like:

```tenantNew :: NewTenant -> ExceptT ServantErr IO Tenant```